crates/shared/src/event_handling.rs

```diff
@@ -1,18 +1,23 @@
 use crate::{current_block::BlockRetrieving, maintenance::Maintaining};
-use anyhow::{Context, Error, Result};
+use anyhow::{ensure, Context, Error, Result};
 use ethcontract::contract::{AllEventsBuilder, ParseLog};
 use ethcontract::errors::ExecutionError;
-use ethcontract::{
-    dyns::DynTransport, BlockNumber as Web3BlockNumber, Event as EthcontractEvent, EventMetadata,
-};
-use futures::{Stream, StreamExt, TryStreamExt};
+use ethcontract::H256;
+use ethcontract::{dyns::DynTransport, Event as EthcontractEvent, EventMetadata};
+use futures::{future, Stream, StreamExt, TryStreamExt};
 use std::ops::RangeInclusive;
 use tokio::sync::Mutex;
 
 // We expect that there is never a reorg that changes more than the last n blocks.
 pub const MAX_REORG_BLOCK_COUNT: u64 = 25;
 // Saving events, we process at most this many at a time.
 const INSERT_EVENT_BATCH_SIZE: usize = 10_000;
+// MAX_BLOCKS_QUERIED is bigger than MAX_REORG_BLOCK_COUNT to increase the chances
+// of avoiding the need for history fetch of block events, since history fetch is less
+// efficient than latest block fetch
+const MAX_BLOCKS_QUERIED: u64 = 2 * MAX_REORG_BLOCK_COUNT;
+
+pub type BlockNumberHash = (u64, H256);
```

Comment:
Should be defined in current_block module. It is confusing that currently the modules depend on eachother.

Reply:
Indeed. Moved.